### PR TITLE
Quote endpoint

### PR DESCRIPTION
**Pull request by eylulyalcinkaya**
I added quote endpoint and its test. I also made the necessary changes in routes.js.

### CODE REVIEW

File: practice-app/backend/controllers/quoteOfTheDay.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    try {
+        Request.send({ //
+            url: 'https://api.forismatic.com/api/1.0/?method=getQuote&lang=en&format=jsonp&jsonp=?',
+            method: 'POST',
+            body: {}
+        }, async function (err, res, body) {
+            if (err || body.error) { // If there is an error respond with the error.
+                return Response.handleError(response, err || body.error);
+            }
+            console.log("Response body: ", body);
```
Comment (eridincu) [inline practice-app/backend/controllers/quoteOfTheDay.js:27]: This line can be deleted when we merge since the code works fine.

Comment (eylulyalcinkaya) [reply practice-app/backend/controllers/quoteOfTheDay.js]: Done!

**Comment by burakcuhadar** [comment]
I got the following error(I think this problem occurs when the quote contains a  \' character):

(node:21718) UnhandledPromiseRejectionWarning: SyntaxError: Unexpected token ' in JSON at position 18
    at JSON.parse (<anonymous>)
    at Request._callback (/home/burak/Masaüstü/cmpe352/bounswe2020group4/practice-app/backend/controllers/quoteOfTheDay.js:30:29)
    at Request.self.callback (/home/burak/Masaüstü/cmpe352/bounswe2020group4/practice-app/backend/node_modules/request/request.js:185:22)
    at Request.emit (events.js:310:20)
    at Request.<anonymous> (/home/burak/Masaüstü/cmpe352/bounswe2020group4/practice-app/backend/node_modules/request/request.js:1154:10)
    at Request.emit (events.js:310:20)
    at IncomingMessage.<anonymous> (/home/burak/Masaüstü/cmpe352/bounswe2020group4/practice-app/backend/node_modules/request/request.js:1076:12)
    at Object.onceWrapper (events.js:416:28)
    at IncomingMessage.emit (events.js:322:22)
    at endReadableNT (_stream_readable.js:1187:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
(node:21718) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1).

**Comment by eylulyalcinkaya** [comment]
> I got the following error(I think this problem occurs when the quote contains a '(apostrophe) character):
> 
> (node:21718) UnhandledPromiseRejectionWarning: SyntaxError: Unexpected token ' in JSON at position 18
> at JSON.parse ()
> at Request._callback (/home/burak/Masaüstü/cmpe352/bounswe2020group4/practice-app/backend/controllers/quoteOfTheDay.js:30:29)
> at Request.self.callback (/home/burak/Masaüstü/cmpe352/bounswe2020group4/practice-app/backend/node_modules/request/request.js:185:22)
> at Request.emit (events.js:310:20)
> at Request. (/home/burak/Masaüstü/cmpe352/bounswe2020group4/practice-app/backend/node_modules/request/request.js:1154:10)
> at Request.emit (events.js:310:20)
> at IncomingMessage. (/home/burak/Masaüstü/cmpe352/bounswe2020group4/practice-app/backend/node_modules/request/request.js:1076:12)
> at Object.onceWrapper (events.js:416:28)
> at IncomingMessage.emit (events.js:322:22)
> at endReadableNT (_stream_readable.js:1187:12)
> at processTicksAndRejections (internal/process/task_queues.js:84:21)
> (node:21718) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1).

I had not encountered this error before, I believe I solved it, @burakcuhadar can you try again?

**Comment by burakcuhadar** [comment]
> > I got the following error(I think this problem occurs when the quote contains a '(apostrophe) character):
> > (node:21718) UnhandledPromiseRejectionWarning: SyntaxError: Unexpected token ' in JSON at position 18
> > at JSON.parse ()
> > at Request._callback (/home/burak/Masaüstü/cmpe352/bounswe2020group4/practice-app/backend/controllers/quoteOfTheDay.js:30:29)
> > at Request.self.callback (/home/burak/Masaüstü/cmpe352/bounswe2020group4/practice-app/backend/node_modules/request/request.js:185:22)
> > at Request.emit (events.js:310:20)
> > at Request. (/home/burak/Masaüstü/cmpe352/bounswe2020group4/practice-app/backend/node_modules/request/request.js:1154:10)
> > at Request.emit (events.js:310:20)
> > at IncomingMessage. (/home/burak/Masaüstü/cmpe352/bounswe2020group4/practice-app/backend/node_modules/request/request.js:1076:12)
> > at Object.onceWrapper (events.js:416:28)
> > at IncomingMessage.emit (events.js:322:22)
> > at endReadableNT (_stream_readable.js:1187:12)
> > at processTicksAndRejections (internal/process/task_queues.js:84:21)
> > (node:21718) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1).
> 
> I had not encountered this error before, I believe I solved it, @burakcuhadar can you try again?

I have tried again. Now it works for most of the quotes containing '(apostrophe), but I got the same error for the following quote:

?({"quoteText":"You\'re not obligated to win. You\'re obligated to keep trying to do the best you can every day.", "quoteAuthor":"Marian Edelman", "senderName":"", "senderLink":"", "quoteLink":"http://forismatic.com/en/789b8b3e83/"})

**Pull request merged**